### ISSUE TITLE
Create no-global-assign rule

### ISSUE THREAD

**Issue by jacksonrayhamilton** [eslint/eslint#5358]
**What version are you using?**

`2.2.0`

**What did you do?**

I have the file `test.js`:

```js
/*global a:false */
a = 5;
```

I have the configuration `.eslintrc.yaml`:

```js
root: true
rules:
  no-undef: 2
```

I linted the file with the command-line argument `eslint test.js`.

**What happened?**

It did not report any errors.

**What did you expect to happen?**

I expect it to warn me that I am assigning to a global variable with the `false` flag, as described [in the documentation](http://eslint.org/docs/user-guide/configuring#specifying-globals).

**Comment by ilyavolodin** [comment]
Verified. I think we removed check for overriding global variables from `no-undef` rule, since it didn't make sense to keep there. But this should be erroring on `no-redeclare` with `builtinGlobals` option on, but it doesn't. So it looks like globals overrides are not captured by anything at this point.

**Comment by michaelficarra** [comment]
It should not be erroring on `no-redeclare`. There's no declaration here.

**Comment by ilyavolodin** [comment]
Ah.. sorry, you are correct. I guess in that case it should trigger `no-native-reassign`. Although this is not a native method... Not sure. @eslint/eslint-team anyone knows which rule should be catching this?

**Comment by BYK** [comment]
My hunch is `no-implicit-globals`

**Comment by gyandeeps** [comment]
Look at scenario 3 here https://github.com/eslint/eslint/issues/5143 and also the comments on this issue.
it might help to diagnose this issue.

**Comment by nzakas** [comment]
Looks like this use case fell through the cracks. I don't know that there's an existing rule where this check makes sense. Do we need `no-global-assign` or something similar?

**Comment by ilyavolodin** [comment]
@nzakas This check used to be in `no-undef`, but was removed from there. I think we should add a new rule. `no-global-assign` sounds reasonable.

**Comment by nzakas** [comment]
Okay, let's go with creating a new `no-global-assign` rule.

**Comment by kaicataldo** [comment]
Actually, it looks like this needs a champion now @eslint/eslint-team

**Comment by mikesherov** [comment]
ok, I'll champion this. Important to maintain feature parity with JSHint here. I recommend:

1. Fixing docs for no-undef, and also have them point to this new rule.
2. Recognizing the `\* global var:true *\ annotation as exceptions to this rule.
3. ignore `\* global var:false *\` as that only applies to `no-undef`. 
4. Also recognizing an array of exceptions to this rule.
5. Rule shall be named `no-global-assign`. 

Given 2 & 3, a dev coming from JSHint SHOULD convert `\* global var: true *\` to `// eslint no-global-assign ["error", ["var"]] *\`, but isn't absolutely required to do so.

**Comment by nzakas** [comment]
Sounds good. As this doesn't seem related to JSCS compat, I removed it from the milestone. We should prioritize the JSCS compat work above this.

**Comment by mikesherov** [comment]
Ok, yeah, I think we can mark this as JSHint compat though, no? And I won't work on this until JSCS compat is done, but I would like to champion this, so when it gets 3 thumbs, I can assign it to myself?

**Comment by nzakas** [comment]
Added a JSHint compat milestone. 

> I would like to champion this, so when it gets 3 thumbs, I can assign it to myself?

Yup, here's one: 👍

**Comment by ilyavolodin** [comment]
Same :+1:

**Comment by zxqfox** [comment]
Okay. 👍

**Comment by jethrolarson** [comment]
I'm covering bugs like this with `"no-shadow": [1, {"builtinGlobals": true}]` but it's awkward that I can't use names like `status`, `name`, etc...

This change should cover the risk better.

**Comment by mysticatea** [comment]
I will work on this.
